R/survey_statistics.r

```diff
@@ -105,9 +107,15 @@ survey_mean <- function(
   }
   prop_method <- match.arg(prop_method)
   if (is.null(df)) df <- survey::degf(cur_svy_full())
-  if (missing(x)) return(survey_prop(vartype = vartype, level = level,
-                                     proportion = proportion, prop_method = prop_method,
-                                     deff = deff, df = df, .svy = cur_svy()))
+  if (missing(x)){
+    if (missing(proportion) & ("ci" %in% vartype)){
```

Comment:
This message is not actually true because of how missing args works (even though (`missing(proportion)` is TRUE, the default value comes from line 97 not line 143 in this branch of code). 

I actually kind of prefer this behavior - my argument would be that this way we're better adhering to the mental model that `survey_mean`'s behavior comes from `survey::svymean` and `survey_prop`'s comes from `survey::svyciprop`. But wanted to check if you disagreed.

My preference is to fix by removing this `inform`, but the other option is to change line 97 so that it `TRUE`.


``` r
library(srvyr)
#> 
#> Attaching package: 'srvyr'
#> The following object is masked from 'package:stats':
#> 
#>     filter
data(api, package = "survey")

dstrata <- apistrat %>%
  as_survey_design(strata = stype, weights = pw)



dstrata %>%
  group_by(above = api99 > 600) %>%
  summarise(
    prop_default = survey_mean(vartype = "ci"),
    prop_true = survey_mean(proportion = TRUE, vartype = "ci"),
    prop_false = survey_mean(proportion = FALSE, vartype = "ci")
  ) %>% select(ends_with("low"))
#> When `proportion` is unspecified, `survey_mean()` now defaults to `proportion = TRUE` when `x` is left out. This should improve confidence interval coverage.
#> This message is displayed once per session.
#> # A tibble: 2 × 3
#>   prop_default_low prop_true_low prop_false_low
#>              <dbl>         <dbl>          <dbl>
#> 1            0.365         0.367          0.365
#> 2            0.483         0.483          0.483
```

Reply:
Sorry, just noticed this as I was getting ready to submit. I need to submit in the next few days, so plan to go with my gut when I get back to this later tonight